crates/ty_python_semantic/src/types.rs

```diff
@@ -6403,11 +6402,17 @@ pub enum DynamicType {
     TodoUnpack,
 }
 
-impl DynamicType {
+#[salsa::tracked]
+impl<'db> DynamicType {
     #[expect(clippy::unused_self)]
     fn normalized(self) -> Self {
         Self::Any
     }
+
+    #[salsa::tracked]
+    fn into_callable(self, db: &'db dyn Db, parameters: Parameters<'db>) -> Type<'db> {
+        CallableType::single(db, Signature::new(parameters, Some(Type::Dynamic(self))))
+    }
```

Comment:
Caching the method definitely makes sense if it improves performance this much, but I think I'd prefer for this to be a method on `CallableType` rather than a method on `DynamicType`. I'm not sure it makes sense to think of this as transforming a `DynamicType` "into a `CallableType`": what we're really doing is creating a `CallableType` that returns a certain `DynamicType`.

Also, could we add a comment saying why we specifically cache the constructor if we're constructing a `CallableType` that returns a `DynamicType`, rather than generally caching the `CallableType::single()` method?